### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by yurushao**
- Add code scanning trigger conditions
- Skip scanning js libraries

### CODE REVIEW

**Review by jparise — COMMENTED** [review]
Could you remove the commented-out blocks that don't apply to this project?

**Comment by yurushao** [comment]
@jparise Thanks. Updated!

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "master" ]
+    # Skip the workflow if PR only contains changes to files matching the following path patterns
+    paths-ignore:
```
Comment (jparise) [inline .github/workflows/codeql-analysis.yml:10]: Should this also skip `tests.py` because that's ignored in `codeql-config.yml`?

**Review by jparise — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (yurushao) [reply .github/workflows/codeql-analysis.yml:10]: Good catch. Yes, we should.

**Pull request merged**